DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/CryptoHashProcessor.cs

```diff
@@ -0,0 +1,91 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text;
+using Dicom;
+using Dicom.IO.Buffer;
+using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class CryptoHashProcessor : IAnonymizerProcessor
+    {
+        private readonly CryptoHashFunction _cryptoHashFunction;
+
+        public CryptoHashProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var cryptoHashSetting = settingFactory.CreateAnonymizerSetting<CryptoHashSetting>(settingObject);
+            _cryptoHashFunction = new CryptoHashFunction(cryptoHashSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            var test = item.GetType().GetMethods();
+
+            var encoding = Encoding.UTF8;
+            if (item is DicomStringElement)
```

Comment:
>item is DicomStringElement [](http://example.com/codeflow?start=16&length=26)

CryptoHashProcessor and EncryptionProcess use item is XXXX to check the type, while other processors use item.valueRepresentation. Why?

Reply:
DicomStringElement is the base class for item.valueRepresentation who has string value.
e.g. Classes for AS, AE, CS...... are derived from DicomStringElement.
Encryption and CryptoHash can process any string values, and we should not compare item.valueRepresentation here.

Reply:
Here is a visio [graph](https://github.com/sowu880/Dicom-demo/blob/main/docs/classes.vsdx) tp show the relationships.